File: Assignment__25/Question1.c
//Accept N numbers from user and return difference between summation of even elements and summation of odd elements.

#include<stdio.h>
#include<stdlib.h>

int EvenOddDifference(int Arr[], int iSize)
{
    int iCnt = 0;
    int iEvenSum = 0;
    int iOddSum = 0;
    int iDfference = 0;

    for(iCnt = 0; iCnt < iSize; iCnt++)    
    {
        if(Arr[iCnt] % 2 == 0)
        {
           iEvenSum = iEvenSum + Arr[iCnt];
        }
    }

    for(iCnt = 0; iCnt < iSize; iCnt++)    
    {
        if(Arr[iCnt] % 2 != 0)
        {
           iOddSum = iOddSum + Arr[iCnt];
        }
    }

    iDfference = iEvenSum - iOddSum;

    if(iDfference < 0)
    {
        iDfference = -iDfference;
    }
    
}

int main()
{
    int iLength = 0;
    int iCnt = 0;
    int *ptr = NULL;
    int iResult = 0;

    printf("Enter the number of elements : ");
    scanf("%d", &iLength);

    ptr = (int *)malloc(iLength * sizeof(int));
    
    if(ptr == NULL)      
     {
        printf("Unable to allocate the memoey\n");
        return -1;
    }

    printf("Enter the elements : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    iResult = EvenOddDifference(ptr,iLength);

    printf("%d is the difference between the summation of even and odd numbers in the given array", iResult);
    
    free(ptr);

    return 0;

}

File: Assignment__25/Question1.java
//Accept N numbers from user and return difference between summation of even elements and summation of odd elements.

import java.util.Scanner;

class ArrayX
{
    public int Arr[];
    public int iSize;

    public ArrayX(int iNo)
    {
        iSize = iNo;

        Arr = new int [iSize];    //Need to remember
    }

    public void Accept()
    {
        Scanner sobj1 = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the numbers : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj1.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("The elements of the array are : ");

        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]);
        }
    }

    public int EvenOddSumDiff()
    {
        int iCnt = 0;
        int iEvenSum = 0;
        int iOddSum = 0;
        int iDifference = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] % 2 == 0)
            {
                iEvenSum = iEvenSum + Arr[iCnt];
            }
            else
            {
                iOddSum = iOddSum + Arr[iCnt];
            } 
        }

        iDifference = iEvenSum - iOddSum;

        if(iDifference < 0)
        {
            iDifference = -iDifference;
        }

        return iDifference;  
    }
}

class Question1
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the number of elements : ");
        int iValue1 = sobj.nextInt();
        int iResult= 0;

        ArrayX aobj = new ArrayX(iValue1);
        aobj.Accept();
        aobj.Display();

        iResult = aobj.EvenOddSumDiff();

        System.out.println(iResult+" is the difference between the sum of even and odd numbers in the given array.");

        aobj = null;
        sobj = null;

        System.gc();
    }

}
